src/features/applications/components/AdminButtonsBar.tsx

```diff
@@ -14,7 +15,11 @@ export const AdminButtonsBar = ({ projectId }: IAdminButtonsBarProps): JSX.Eleme
   const isAdmin = useIsAdmin();
   const { isCorrectNetwork, correctNetwork } = useIsCorrectNetwork();
 
-  const approve = useApproveApplication({});
+  const approve = useApproveApplication({
+    onSuccess: () => {
+      window.location.reload();
```

Comment:
It's not a good practice to reload page on action/success/error.

Reply:
Agree, but I don't have better way to call fetch immediately 🤔
How about I made the `refetchInterval` to 10 seconds like now?

Reply:
This lib has something for immediate refetching values, the docs should have this.

Reply:
So what I wanna do is to refetch instantly in `onSuccess`

Reply:
Right, but make sure you don't call it in a loop, so if there are no new data, just skip refetching.

Reply:
I tried several ways, no best way to solve this problem without reloading page now, would think over it tomorrow.

Reply:
I tried on calling only the `refetch` instead of setting the `refetchInterval`, but it's not working, so I keep this version until the best solution is found.

Reply:
changed from `onSuccess` to check `approve.isSuccess` and call `refetch` in `useEffect` hook, however still not working QQ

Reply:
Let's create issue for this and todo with a link for this issue.
The solution here is to merge cached values with new data, but I haven't worked with trpc stack closely so I don't know how to merge the data.
If this task is not urgent, you can try to research this one.

Reply:
Of course the ultimate solution here is to merge the cached data and new data, and the question remains here is "how to get the new data", this worth a new issue.

But I think this PR solves an important problem that should be solved in v1.
So I'm thinking about could we have a temporary solution here for smooth user experience, and delve into the best solution afterwards?  what should the temporary solution be like is my concern QQ

Reply:
@0xmad how do you think I changed to something like this now? (but I think the data is not updated right after `isSuccess` so should call after something like 3 seconds? the `isSuccess` is too fast which is earlier than the transaction succeeded QQ)

Reply:
> @0xmad how do you think I changed to something like this now?
Can you show what you mean?

Reply:
so now I split some functions out the `fetchAttestation.ts` file, and instead of setting `cacheInterval` to 0 for uncached fetch, made another file for normal fetch, since the `node-fetch-cache` package cannot be imported or called in the browser. Once the application is approved, the `useEffect` hook would call normal fetch and merge the results with the original fetch result. 

But if it refetch once it's `isSuccess`, the attestation is actually not yet updated, so I made it sleep for 5 seconds and refetch.

Reply:
I think it's good as a temporary solution.

Reply:
open an issue for it: https://github.com/privacy-scaling-explorations/maci-rpgf/issues/229